.github/workflows/ci.yml

```diff
@@ -93,6 +95,10 @@ jobs:
         !contains(matrix.platform.target, 'wasm32'))
       run: cargo $CMD test --verbose --target ${{ matrix.platform.target }} $OPTIONS --features $FEATURES
 
+    - name: Lint with clippy
+      shell: bash
+      if: (matrix.rust_version != 'nightly') && !contains(matrix.platform.options, '--no-default-features')
+      run: cargo clippy --all-targets --target ${{ matrix.platform.target }} $OPTIONS --features $FEATURES
```

Comment:
I think we don't want to run clippy for examples actually.

Reply:
I think we should be consistent and enforce it everywhere, so that the examples are also clean-enough.

Reply:
In addition there's `--all-targets` here and you fixed+allowed clippy lints in the `examples/`, so this is only sensible?

Reply:
I'd add `-D warnings` on CI then, I guess.

Having deny in lib.rs is still easier to understand that you must fix those issues.

Reply:
Yeah, leave the deny in `lib.rs`, that should hopefully decrease the amount of noise in PRs that would otherwise require lots of cleanup pushes to "make the CI happy".

Reply:
I'm fairly certain that the general advice here is to _not_ deny warnings in `*.rs` files, and instead only do it in CI, as it might break compilation with future versions of `rustc` and/or `clippy`.

Reply:
That's the point, we don't deny warnings for rustc or any related command, just when running clippy. So there's no way the build will break, only clippy could, but it would only mean that CI is broken.

Reply:
Ah, I see. That's clever.